narwhal/config/src/committee.rs

```diff
@@ -0,0 +1,512 @@
+// Copyright (c) 2021, Facebook, Inc. and its affiliates
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{CommitteeUpdateError, ConfigError, Epoch, Stake};
+use crypto::{NetworkPublicKey, PublicKey, PublicKeyBytes};
+use fastcrypto::traits::EncodeDecodeBase64;
+use mysten_network::Multiaddr;
+use mysten_util_mem::MallocSizeOf;
+use once_cell::sync::OnceCell;
+use rand::rngs::StdRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashSet};
+use std::fmt::{Display, Formatter};
+
+#[derive(Clone, Serialize, Deserialize, Debug, Eq, PartialEq)]
+pub struct Authority {
+    /// The id under which we identify this authority across Narwhal
+    #[serde(skip)]
+    id: OnceCell<AuthorityIdentifier>,
```

Comment:
Why does the `id` field not always set? And how does it get set?

Reply:
This is set via the `Committee` when we provide the authorities. We currently have two ways to initialise a committee:
* either programmatically 
* via a file (when running as standalone)

when reading the committee from a file the id doesn't/shouldn't exist there - it should be committee's responsibility to come up with whatever is considered appropriate (at least from my perspective). Thus I've put this behind a `OnceCell` and skipped serialise/deserialise.

Reply:
Sorry the link is not working for me. Can we copy out the value from `Option` instead? My only concern is that OnceCell has some synchronization overhead and is only needed for dynamical initializations. Regardless, I trust your judgement here.